UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs

```diff
@@ -310,6 +274,24 @@ public virtual void Awake()
 		registerTile = GetComponent<RegisterTile>();
 		rotatable = GetComponent<Rotatable>();
 		pickupable.DirectSetComponent(GetComponent<Pickupable>());
+		SetRotationTarget();
+	}
+
+	private void SetRotationTarget()
+	{
+		if (rotationTarget != null) return;
+		if (this is MovementSynchronisation c && c.playerScript.RegisterPlayer.LayDownBehavior != null)
```

Comment:
base class shouldn't handle cases for a child class. Can you make this method protected virtual and do the second part of the if on the subclass?

Reply:
nevermind, I just took a look at this class and the subclass. I wouldn't want to touch any of that either.